pkg/skaffold/schema/validation/validation_test.go

```diff
@@ -750,3 +752,123 @@ func TestValidateLogsConfig(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateArtifactCircularDependencies(t *testing.T) {
+	tests := []struct {
+		description string
+		artifactLen int
+		dependency  [][]int
+		shouldErr   bool
+	}{
+		{
+			description: "artifacts with no dependency",
+			artifactLen: 5,
+		},
+		{
+			description: "artifacts with dependencies",
+			dependency: [][]int{
+				{1, 3, 4},
+				{3, 2},
+				{2, 4},
+				{4, 5},
+			},
+			artifactLen: 5,
+		},
+		{
+			description: "artifacts with circular dependencies",
+			dependency: [][]int{
+				{1, 3, 4},
+				{3, 2},
+				{2, 1},
+				{4, 5},
+			},
+			artifactLen: 5,
+			shouldErr:   true,
+		},
+		{
+			description: "0 artifacts",
+			artifactLen: 0,
+		},
+	}
+	for _, test := range tests {
+		testutil.Run(t, test.description, func(t *testutil.T) {
+			artifacts := make([]*latest.Artifact, test.artifactLen)
+			for i := 0; i < test.artifactLen; i++ {
+				a := fmt.Sprintf("artifact%d", i+1)
+				artifacts[i] = &latest.Artifact{ImageName: a}
+			}
+
+			setDependencies(artifacts, test.dependency)
+			errs := validateArtifactCircularDependencies(artifacts)
+			expected := []error{
+				fmt.Errorf("cycle detected in build dependencies involving 'artifact1'"),
+			}
+			if test.shouldErr {
+				t.CheckDeepEqual(expected, errs, cmp.Comparer(errorsComparer))
+			} else {
+				t.CheckDeepEqual(0, len(errs))
+			}
+		})
+	}
+}
+
+func setDependencies(a []*latest.Artifact, d [][]int) {
+	for _, dep := range d {
+		for i := range dep {
+			if i == 0 {
+				continue
+			}
+			a[dep[0]-1].Dependencies = append(a[dep[0]-1].Dependencies, &latest.ArtifactDependency{
+				ImageName: a[dep[i]-1].ImageName,
+			})
+		}
+	}
+}
+
+func TestValidateArtifactDependencyUniqueness(t *testing.T) {
+	artifacts := []*latest.Artifact{
+		{
+			ImageName: "artifact1",
+			Dependencies: []*latest.ArtifactDependency{
+				{
+					Alias:     "alias2",
+					ImageName: "artifact2a",
+				},
+				{
+					Alias:     "alias2",
+					ImageName: "artifact2b",
+				},
```

Comment:
These take up a lot of space and could be shortened to just:
```suggestion
				{Alias:     "alias2", ImageName: "artifact2a"},
				{Alias:     "alias2", ImageName: "artifact2b"},
```